### PR TITLE
Minimal Seed Data for Admin Onboarding

### PR DESCRIPTION
**Pull request by Ridhwana**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
When we created a new node for the E2E testing, we assumed that we would need no seed data. However, after some more investigation, it turns out that we may need some minimal seed data that will be used for configuration, like the Feature Flags and possibly in the future some SMTP settings that control redirects.

Hence, in this PR I've changed the "non seeded" flow to be a "minimal seeded flow".

**NOTE: If we think that the naming convention may cause confusion, I'm open to suggestions :)**

## Related Tickets & Documents
https://github.com/forem/forem/pull/14259

Enhancement of https://github.com/forem/forem/pull/14285

## QA Instructions, Screenshots, Recordings
Run `yarn e2e:minimal-seed`.

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [ ] No, This simply amend the naming for the testing infrastructure.
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [x] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Review by rhymes — COMMENTED** [review]
I wonder if this should simply be named "seed_onboarding" or something like that, as the concept of "minimal" is entirely context dependent and people will eventually add things to a file called "minimal seed".

If we instead are explicit about that being the seed file for onboarding, people will only use it for that

**Comment by Ridhwana** [comment]
> I wonder if this should simply be named "seed_onboarding" or something like that, as the concept of "minimal" is entirely context dependent and people will eventually add things to a file called "minimal seed".
> 
> If we instead are explicit about that being the seed file for onboarding, people will only use it for that

You're right, this is very subjective. I did think about being explicit about it and make it specific to the onboarding, but I wasn't sure if we would only use it for onboarding. But maybe better to be specific that too general. I'll hear out other suggestions too and then change accordingly. Thanks @rhymes!

cc @nickytonline

File: db/seeds.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 ##############################################################################
 # Default development settings are different from production scenario
-
+FeatureFlag.enable(:profile_admin)
```
Comment (nickytonline) [inline db/seeds.rb:17]: Question (potentially blocking): This enables the new admin onboarding, but the local development DB will be seeded with users, so the new admin onboarding would never be seen no?

File: package.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
     "test:watch": "jest app/javascript/ bin/ --watch",
     "postcss": "postcss public/assets/*.css -d public/assets 2> postcss_error.log",
     "e2e": "bin/e2e",
-    "e2e:noseed": "yarn run e2e --no-seed"
+    "e2e:minimal-seed": "yarn run e2e --minimal-seed"
```
Comment (nickytonline) [inline package.json:28]: Nitpick (non-blocking): Consider a shorter command name.

```suggestion
    "e2e:minseed": "yarn run e2e --minimal-seed"
```

File: docs/tests/e2e-tests.md [hunk 3]
```diff
@@ -44,8 +44,8 @@ coordinate all this work.
 ### 1. From the command line, run `yarn e2e`
 
 Note: If you want to run E2E tests that do not require seeded data, run
-`yarn e2e:noseed`. Some tests, like the admin onboarding require there to be no
-seeded data.
+`yarn e2e:minimal-seed`. Some tests, like the admin onboarding require there to
```
Comment (nickytonline) [inline docs/tests/e2e-tests.md:47]: Nitpick (non-blocking): Consider a shorter command name.

```suggestion
`yarn e2e:minseed`. Some tests, like the admin onboarding require there to
```

**Review by nickytonline — CHANGES_REQUESTED** [review]
This is looking good @Ridhwana. I left some suggestions and a change request. 😎

**Comment by nickytonline** [comment]
> > I wonder if this should simply be named "seed_onboarding" or something like that, as the concept of "minimal" is entirely context dependent and people will eventually add things to a file called "minimal seed".
> > If we instead are explicit about that being the seed file for onboarding, people will only use it for that
> 
> You're right, this is very subjective. I did think about being explicit about it and make it specific to the onboarding, but I wasn't sure if we would only use it for onboarding. But maybe better to be specific that too general. I'll hear out other suggestions too and then change accordingly. Thanks @rhymes!
> 
> cc @nickytonline

Given, we only are using it for admin onboarding right now, @rhymes suggestion is a good one. If in the future we used it for more than that, we can always rename it.

File: db/seeds.rb (hunk 1 above)
Comment (Ridhwana) [reply db/seeds.rb:17]: @nickytonline I don't for see this causing any issues, since there are no tests associated with the minimal seeded flow. It was just the prep work for https://github.com/forem/forem/pull/14259. I can remove it and leave the file blank and add it to #14259. Thoughts?

File: config/initializers/cypress_rails.rb [hunk 4]
```diff
@@ -8,8 +8,11 @@
   # Called once, before either the transaction or the server is started
   puts "Starting up server for end to end tests."
 
-  if ENV["SKIP_SEED_DATA"].blank?
-    Rails.application.load_tasks
+  Rails.application.load_tasks
+
+  if ENV["MINIMAL_SEED_DATA"]
+    Rake::Task["db:seed:minimal"].invoke
```
[the remark this answers is not in the document]
Comment (Ridhwana) [reply config/initializers/cypress_rails.rb:14]: @nickytonline shall we rename this e2e_onboarding then based on your final comment below?

File: package.json (hunk 2 above)
Comment (Ridhwana) [reply package.json:28]: @nickytonline and this one `e2e:onboardingseed`?

File: db/seeds.rb (hunk 1 above)
Comment (nickytonline) [reply db/seeds.rb:17]: It makes sense to add it to #14259. I'm not sure what you mean by leave the file blank though, as this is the seed file for local development.

File: package.json (hunk 2 above)
Comment (Ridhwana) [reply package.json:28]: I'm not sure what we can shorten this to, maybe `onbseed`, although this isnt very semantic so I would prefer the fullname or even one with the `-`.

File: config/initializers/cypress_rails.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (nickytonline) [reply config/initializers/cypress_rails.rb:14]: Maybe `e2e_admin_onboarding`?

File: package.json (hunk 2 above)
Comment (nickytonline) [reply package.json:28]: Yeah, now that we'll be calling it admin onboarding for the flows folder, go ahead with a longer name to make it clear.

File: db/seeds.rb (hunk 1 above)
Comment (Ridhwana) [reply db/seeds.rb:17]: Ok so I looked at this commit again and realized I copied this incorrectly before I committed 🤦🏽‍♀️ (hence your comment above and the confusion from myside ). 

It was meant to be 
```
FeatureFlag.enable(:creator_onboarding)
```
and not `:profile_admin`.

In #14259 I will be introducing the Cypress tests that will depend on this `creator_onboarding` feature flag being in place. 

Now that I'm on the same page - I'd actually like to leave the flag in the seed file to prep it for the next PR, because we are renaming files anyways to indicate that we'll be testing the onboarding flow.

Comment (Ridhwana) [reply db/seeds.rb:17]: Ignore whatever I said, this commit needs to be removed 🤦🏽‍♀️ This commit was a mistake and was just a test on the normal `seeds.rb`. I'm clearly very tired and have been working with too many PR's today 😅

Comment (nickytonline) [reply db/seeds.rb:17]: 😅

**Comment by Ridhwana** [comment]
Changes implemented. Thanks @nickytonline and @rhymes for the suggestions!

**Review by rhymes — APPROVED** [review]
🆙

**Review by nickytonline — APPROVED** [review]
Thanks for making the changes @Ridhwana! 🚢
